graphql/schemabuilder/pagination.go

```diff
@@ -393,39 +396,33 @@ func (c *connectionContext) pagesFromEdges(edges []Edge, limit int) (pages []str
 	return pages
 }
 
-func (c *connectionContext) applyTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs) ([]interface{}, error) {
-	if args.FilterText == nil || *args.FilterText == "" {
-		return nodes, nil
-	}
-
-	nodesToKeep := make([]bool, len(nodes))
+type SafeBatchNodesToKeep struct {
+	nodesToKeep []bool
+	mux         sync.Mutex
+}
 
+func (c *connectionContext) applyBatchTextFilter(ctx context.Context, nodes []interface{}, args PaginationArgs, batchedFields map[string]*graphql.Field) ([]bool, error) {
```

Comment:
instead of passing in the paginationArgs let's pass in the `matchStrings` directly.